benchmarking/evaluations/metrics/utils/team_option.py

```diff
@@ -0,0 +1,43 @@
+from benchmarking.evaluations.metrics.utils.team_calculations import *
+
+
+class TeamOption:
```

Comment:
nit;

This is super cool but feels kinda overkill imo? Like why not just pass a reference to the function you want to the AverageSocialSatisfaction directly?

i.e.
```python
AverageSocialSatisfaction(social_function=is_strictly_happy_team_enemy)
```

Reply:
Why couldn't I think of this 😭